### PR TITLE
Check for memory overflow during tensor allocation

### PR DESCRIPTION
**Pull request by joyalbin**

### CODE REVIEW

**Comment by joyalbin** [comment]
@aselle Could you please review this PR

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/lite/core/subgraph.cc [hunk 1]
```diff
@@ -424,44 +424,69 @@ TfLiteStatus Subgraph::CheckTensorIndices(const char* label, const int* indices,
...
+  // (see section 4.7 [conv.integral] of the C++14 standard).
+  const uint64_t ux = x;
+  const uint64_t uy = y;
+  const uint64_t uxy = ux * uy;
+
+  // Check if we overflow uint64, using a cheap check if both inputs are small
+  if (((ux | uy) >> 32) != 0) {
+    // Ensure nonnegativity.  Note that negative numbers will appear "large"
+    // to the unsigned comparisons above.
+    TF_LITE_ENSURE(context_, x >= 0 && y >= 0);
```
Comment (jdduke) [inline tensorflow/lite/core/subgraph.cc]: I'm not sure we should return the error as an int64_t here? As that will be ignored?

Either this method should return an error, and the int64_t result is a pointer arg, or we should return -1 here.

**Review by jdduke — CHANGES_REQUESTED** [review]

**Comment by rthadur** [comment]
@joyalbin can you please check review comments ?

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Comment by aselle** [comment]
Thanks for the submission. I've reworked this internally based on some refactors and also optimized the code from overflow.h to not have to do some checks since we already know it is unsigned this should land soon.

**Pull request closed without merging**